annotation-statistics/src/main/java/eu/europeana/annotation/statistics/service/AnnotationStatisticsService.java

```diff
@@ -0,0 +1,93 @@
+    	//getting the overall annotations statistics
+    	annoFacetStatsJson = solrService.getAnnotationStatisticsJsonFacets(); 
+    	extractAnnotationStatisticsOverallFromJsonFacet(annoFacetStatsJson, annoMetric);
+    }
+    
+    private void extractAnnotationStatisticsClientUserFromJsonNestedFacet (QueryResponse annoFacetStats, AnnotationMetric annoMetric, String facetField, String target) {
+    	List<AnnotationStatistics> annotationStatistics = new ArrayList<AnnotationStatistics>();
+	    annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(facetField).getBuckets();
+    	for (BucketJsonFacet mainBucket : annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(facetField).getBuckets()) {
+    		AnnotationStatistics annoStats = null;
+    		if(target.compareToIgnoreCase(AnnotationStatisticsConstants.CLIENT)==0) {
+    			annoStats = new AnnotationStatisticsClient();
+    			((AnnotationStatisticsClient)annoStats).setClient(mainBucket.getVal().toString());
+    		}
+    		else if(target.compareToIgnoreCase(AnnotationStatisticsConstants.USER)==0) {
+    			annoStats = new AnnotationStatisticsUser();
+    			((AnnotationStatisticsUser)annoStats).setUser(mainBucket.getVal().toString());
+    		}
+    		
+    		if (annoStats==null) continue;
+		    	
+    		setAnnotationStatisticsFromJsonBuckets(annoStats, mainBucket.getBucketBasedFacets(SolrAnnotationConstants.SCENARIO).getBuckets());
+        	annotationStatistics.add(annoStats);
+    	}
+	    
+		if(target.compareToIgnoreCase(AnnotationStatisticsConstants.CLIENT)==0) {
+			annoMetric.setAnnotationStatisticsClients(annotationStatistics);
+		}
+		else if(target.compareToIgnoreCase(AnnotationStatisticsConstants.USER)==0) {
+			annoMetric.setAnnotationStatisticsUsers(annotationStatistics);
+		}
+    }
+    
+    private void extractAnnotationStatisticsOverallFromJsonFacet (QueryResponse annoFacetStats, AnnotationMetric annoMetric) {
+    	AnnotationStatistics annotationStatisticsOverall = new AnnotationStatistics();	
+    	setAnnotationStatisticsFromJsonBuckets(annotationStatisticsOverall,annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(SolrAnnotationConstants.SCENARIO).getBuckets());
+    	annoMetric.setAnnotationStatisticsScenarios(annotationStatisticsOverall);
+    }
+    
+    private void setAnnotationStatisticsFromJsonBuckets (AnnotationStatistics annoStats, List<BucketJsonFacet> facetJsonBuckets) {
+    	for (BucketJsonFacet facetJsonBucket : facetJsonBuckets) {  
+        	if(facetJsonBucket.getVal().toString().compareToIgnoreCase(AnnotationScenarioTypes.GEO_TAG)==0) {
+        		annoStats.setGeoTag(facetJsonBucket.getCount());
+        	}
+        	if(facetJsonBucket.getVal().toString().compareToIgnoreCase(AnnotationScenarioTypes.TRANSCRIPTION)==0) {
+        		annoStats.setTranscription(facetJsonBucket.getCount());
+        	}
+        	if(facetJsonBucket.getVal().toString().compareToIgnoreCase(AnnotationScenarioTypes.OBJECT_LINK)==0) {
```

Comment:
please switch the position of the constant and use equalsIgnoreCase

Reply:
This code is not needed any more.

Reply:
Improved.